### DISCUSSION TITLE
Trying to use nwipe.config and nwipe_customers from a FTP

### DISCUSSION THREAD

**Discussion by Julien2v2c** [PartialVolume/shredos.x86_64#247] [General]
Hello, 

I am a new ShredOs user and I managed to configure the soft from an USB to send reports and log to a ftp server. 
By looking into the nwipe_laucher I found the "shredos_config" option that should let me retrive the nwipe.config and customers files from my ftp server. 
But it doesn't work, when I try booting from my USB and I go to check organization, the file used is the one from the USB key not from my ftp server.

Is there a way to use this option ?

This is my grub.cfg :
```
set default="0"
set timeout="0"

menuentry "shredos" {
	linux /boot/shredos console=tty3 loglevel=3 shredos_exclude_boot_disc="yes" nwipe_options="--nousb" loadkeys=fr shredos_config="ftp:192.168.1.130:/data/:shredos:shredos" lftp="open 192.168.1.130; user shredos shredos; cd data; mput nwipe_*.txt; mput nwipe_report_*.pdf"
}
```

**Comment by PartialVolume** [comment]
> Hello,
> 
> I am a new ShredOs user and I managed to configure the soft from an USB to send reports and log to a ftp server. By looking into the nwipe_laucher I found the "shredos_config" option that should let me retrive the nwipe.config and customers files from my ftp server. But it doesn't work, when I try booting from my USB and I go to check organization, the file used is the one from the USB key not from my ftp server.
> 
> Is there a way to use this option ?
> 
> This is my grub.cfg :
> 
> ```
> set default="0"
> set timeout="0"
> 
> menuentry "shredos" {
> 	linux /boot/shredos console=tty3 loglevel=3 shredos_exclude_boot_disc="yes" nwipe_options="--nousb" loadkeys=fr shredos_config="ftp:192.168.1.130:/data/:shredos:shredos" lftp="open 192.168.1.130; user shredos shredos; cd data; mput nwipe_*.txt; mput nwipe_report_*.pdf"
> }
> ```

Because of the config you are using it looks like you are building from source from the latest master, if you could confirm that. Can you also post the contents of /transfer.log. Thanks

**Comment by PartialVolume** [comment]
Before you grab the transfer.log can you also append a `:d` to `shredos_config="ftp:192.168.1.130:/data/:shredos:shredos"` so it becomes `shredos_config="ftp:192.168.1.130:/data/:shredos:shredos:d"`. This will make the transfer.log verbose and will show ftp errors.

Also depending on how your ftp server is configured you may find you have to specify the full path .i.e `shredos_config="ftp:192.168.1.130:/home/username/data/:shredos:shredos:d"`

**Comment by Julien2v2c** [comment]
Hi, 

Thanks a lot for your answer. I am actually using the latest .img so the shredos_config option is probably not yet implemented in this version, I compiled from source yesterday and I will try probably Monday. I am sorry, It's the first time I use github and I thought the  .img and the git repository where at the same developpement level, which is not. Anyway I will try from the source build next week and see how it goes.

**Comment by PartialVolume** [comment]
>shredos_config option is probably not yet implemented in this version

Yes, that's correct. Let me know if you run into any issues but the FTP should work fine for restoring the config & customer files from your server.

**Comment by Julien2v2c** [comment]
Hello,

So I make some test after making the image from sources and it works very well to get the config from ftp. I had to put config file as read only because it seems that when wipe is finished it tries to rewrite the config file to the server.

I have one problem with a motherboard GA-78LMT-USB3 R2, when I try to load config file it can't get it from the server even if reports are well saved on the ftp server.

When I do an ALT+F2 -> more trasfert.log, the first ping (before any wipe) can't reach the network. But when, on the same prompt, i ping my server it responds well. I think that's an issue with the motherboard but I can't figure out why.

Edit : I added a sleep 5 before the first ping in nwipe_launcher then I rebuild and it seems to work now, (I am still testing). Maybe the Lan chipset didn't get enough time to initialize

**Reply by PartialVolume** [reply]
Yes, you are correct, this was previously reported by somebody else I need to add a sleep loop to the launcher that launches nwipe only after the network is up and the server is responding to a ping. I'll let you know once I've added the change and perhaps you could test it for me?, now you can build from source.

**Reply by Julien2v2c** [reply]
Hello, 

Sorry for the delay of my answer.

Yes I will be able to make some tests for you, when changes will be done.

**Reply by PartialVolume** [reply]
Thanks, I've made the changes to the launcher and have been testing for both ftp & tftp, all looks good. I'll commit in the next day or two and then do a official release with .imgs and Isos for testing.
